### PR TITLE
Update List Command 

### LINKED ISSUE #86: Add list by client or vendor

### PR DESCRIPTION
**Pull request by Teddayz**
Fixes #86

### CODE REVIEW

File: src/main/java/seedu/ddd/logic/parser/ListCommandParser.java [hunk 1]
```diff
@@ -21,17 +31,29 @@ public class ListCommandParser implements Parser<ListCommand> {
...
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_TAG, PREFIX_ID);
+        String preambleString = argMultimap.getPreamble();
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            Set<Tag> tagSet = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+            return new ListCommand(new ContactContainsTagPredicate(tagSet));
+        } else if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String trimmedArgs = argMultimap.getValue(PREFIX_NAME).get().trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.NAME_MESSAGE_USAGE));
```
Comment (casperngeen) [inline src/main/java/seedu/ddd/logic/parser/ListCommandParser.java:45]: Is it possible to abstract this if-else block into something simpler?

File: src/main/java/seedu/ddd/logic/commands/ListCommand.java [hunk 2]
```diff
@@ -19,11 +20,18 @@ public class ListCommand extends Command {
 
     public static final String MESSAGE_SUCCESS = "Listed all persons";
```
Comment (leslieyip02) [inline src/main/java/seedu/ddd/logic/commands/ListCommand.java:21]: Should change this message to "Listed all contacts".

File: src/main/java/seedu/ddd/logic/commands/ListCommand.java [hunk 3]
```diff
@@ -17,13 +17,20 @@ public class ListCommand extends Command {
...
+    public static final String TAG_MESSAGE_USAGE = COMMAND_WORD + " " + PREFIX_TAG
+            + " : All contacts that match any of "
             + "the specified tags (case-insensitive) will be displayed as a list with index numbers.\n"
             + "Parameters: " + PREFIX_TAG + " TAG...\n"
             + "Example: " + COMMAND_WORD + " " + PREFIX_TAG + " catering " + PREFIX_TAG + " photography";
+    public static final String NAME_MESSAGE_USAGE = COMMAND_WORD + " " + PREFIX_NAME + " : All contacts with the "
+            + "specified name (case-insensitive) will be displayed as a list with index numbers. \n"
+            + "Parameters: " + PREFIX_NAME + "NAME...\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_NAME + " jane ";
+
```
Comment (casperngeen) [inline src/main/java/seedu/ddd/logic/commands/ListCommand.java:33]: Can you change this to a general message since now we can find by all commands?

**Comment by casperngeen** [comment]
LGTM

**Pull request merged**